Documentation/gitlab-ci.md

```diff
@@ -0,0 +1,260 @@
+# GitLab CI/CD Setup Guide
+
+This document provide a summary of the GitLab documentation, sufficient to setting
+up and configuring a new GitLab repository with self-hosted build workers 
+(`gitlab-runner`s). This document will only cover a small portion of the 
+official documentation - just enough to setup a fresh repository from scratch.
+
+## Local Setup
+
+For the sake of not messing about, this document will use a fresh GitLab server
+instance for demoing. To bring up a fresh docker instance with a GitLab server
+you can run:
```

Comment:
I would add a note here clarifying that we use a hosted GitLab server at gitlab.com, so that this local setup is purely for demo purposes and to better understand the application.